### ISSUE TITLE
Take platform screenshots on Android

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#152189]
`FlutterDriver` already supports [taking a screenshot](https://api.flutter.dev/flutter/flutter_driver/FlutterDriver/screenshot.html), but this screenshot is based on the Flutter engine's view of the world, and as a result, omits platform-views (views that are drawn and managed by the platform).

An example of platform views include:

- web-views, such as [webview_flutter_android](https://github.com/flutter/packages/tree/main/packages/webview_flutter/webview_flutter_android)
- google maps, such as [google_maps_flutter_android](https://github.com/flutter/packages/tree/main/packages/google_maps_flutter/google_maps_flutter_android)
- google ads, such as [google_mobile_ads](https://github.com/googleads/googleads-mobile-flutter)

Using a new API (e.g. `nativeDriver.screenshot()`), it should be possible to take a "native" screenshot, using whatever the platform provides. For example, on Android, this would likely be (a variation of) `adb shell screencap`, on iOS `idevicescreenshot`, and so on.

These screenshots will not be cropped, and may include system shell UI. The former will not be addressed at the moment, but the latter can be partially addressed by using a device's appropriate "demo mode" (sometimes called "immersive mode"), or otherwise trying to create a somewhat hermetic screenshot environment (to the best of our ability).

Platform screenshots will likely be _opaque_ objects, because they will exist as a file on disk, not bytes.

**Comment by ueman** [comment]
This seems related to https://github.com/flutter/flutter/issues/102866, where the functionality of taking "screenshot"s via RepaintBoundary for PlatformViews is requested. If that's done, it would also resolve this, wouldn't it?

**Comment by matanlurey** [comment]
> This seems related to #102866, where the functionality of taking "screenshot"s via RepaintBoundary for PlatformViews is requested. If that's done, it would also resolve this, wouldn't it?

I don't think so, all this is doing is adding a `flutter_driver`-specific API for using the platform screenshot mechanism.
